### PR TITLE
add orc format

### PR DESCRIPTION
**Pull request by yaojiefeng**

### CODE REVIEW

**Comment by maochf** [comment]
Have you tested it on Azure?

File: adls-plugins/src/main/java/co/cask/hydrator/plugin/sink/StructuredToOrcTransformer.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  @Override
+  public OrcStruct transform(StructuredRecord input, Schema schema) {
+    List<Schema.Field> fields = input.getSchema().getFields();
+    OrcStruct orcRecord = parseOrcSchema(input.getSchema());
+    //populate ORC struct orcRecord object
+    for (int i = 0; i < fields.size(); i++) {
+      Schema.Field field = fields.get(i);
+      try {
+        WritableComparable writable = convertToWritable(field, input);
+        orcRecord.setFieldValue(fields.get(i).getName(), writable);
```
Comment (maochf) [inline adls-plugins/src/main/java/co/cask/hydrator/plugin/sink/StructuredToOrcTransformer.java]: field.getName()

**Comment by yaojiefeng** [comment]
I have tested it on sdk and singlenode cluster, is there any concern on Azure?

File: adls-plugins/src/main/java/co/cask/hydrator/plugin/sink/StructuredToOrcTransformer.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        throw new IllegalArgumentException(String.format("Not a valid Schema %s", inputSchema.toString()), e);
+      }
+      TypeDescription schema = TypeDescription.fromString(builder.toString());
+      orcRecord = (OrcStruct) OrcStruct.createValue(schema);
+      schemaCache.put(inputSchema, schema);
+    }
+    return orcRecord;
+  }
+
+  private WritableComparable convertToWritable(Schema.Field field, StructuredRecord input)
```
Comment (anwar6953) [inline adls-plugins/src/main/java/co/cask/hydrator/plugin/sink/StructuredToOrcTransformer.java]: this logic in this method seems quite similar to other code in hydrator-plugins repo

Comment (yaojiefeng) [reply adls-plugins/src/main/java/co/cask/hydrator/plugin/sink/StructuredToOrcTransformer.java]: Yeah, this is from https://github.com/caskdata/hydrator-plugins/blob/develop/core-plugins/src/main/java/co/cask/hydrator/plugin/common/StructuredToOrcTransformer.java

File: adls-plugins/docs/ADLSBatchSink-batchsink.md [hunk 3]
```diff
@@ -19,9 +19,9 @@ Properties
 **fileSystemProperties:** A JSON string representing a map of properties
 needed for the distributed file system. (Macro-enabled)
 
-**outputFormat:** The format of output files. Must be 'avro' or 'text'.
+**outputFormat:** The format of output files. Must be 'avro', 'text' or 'orc'.
 
-**schema:** Output schema of the JSON document. Required for avro output format. If left empty for text output format, the schema of input records will be used. This must be a subset of the schema of input records. Fields of type ARRAY, MAP, and RECORD are not supported with the text format. Fields of type UNION are only supported if they represent a nullable type.
+**schema:** Output schema of the JSON document. Required for avro and orc output format. If left empty for text output format, the schema of input records will be used. This must be a subset of the schema of input records. Fields of type ARRAY, MAP, and RECORD are not supported with the text format. Fields of type UNION are only supported if they represent a nullable type.
```
Comment (albertshau) [inline adls-plugins/docs/ADLSBatchSink-batchsink.md]: format -> formats. 'and orc output formats'

File: adls-plugins/src/main/java/co/cask/hydrator/plugin/sink/StructuredToOrcTransformer.java (hunk 2 above)
Comment (albertshau) [reply adls-plugins/src/main/java/co/cask/hydrator/plugin/sink/StructuredToOrcTransformer.java]: I believe this is copied in other plugin projects as well. Can you open a jira to move common orc functions to some common lib module that all these projects can add as a dependency.

Comment (yaojiefeng) [reply adls-plugins/src/main/java/co/cask/hydrator/plugin/sink/StructuredToOrcTransformer.java]: Created the JIRA for the improvement: https://issues.cask.co/browse/CDAP-13076

**Pull request merged**